includes/class-autoloader.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Autoloader for Activitypub.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub;
+
+/**
+ * An Autoloader that respects WordPress's filename standards.
+ */
+class Autoloader {
+
+	/**
+	 * Namespace separator.
+	 */
+	const NS_SEPARATOR = '\\';
+
+	/**
+	 * The prefix to compare classes against.
+	 *
+	 * @var string
+	 * @access protected
+	 */
+	protected $prefix;
+
+	/**
+	 * Length of the prefix string.
+	 *
+	 * @var int
+	 * @access protected
+	 */
+	protected $prefix_length;
+
+	/**
+	 * Path to the file to be loaded.
+	 *
+	 * @var string
+	 * @access protected
+	 */
+	protected $path;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $prefix Namespace prefix all classes have in common.
+	 * @param string $path   Path to the files to be loaded.
+	 */
+	public function __construct( $prefix, $path ) {
+		$this->prefix        = $prefix;
+		$this->prefix_length = strlen( $prefix );
+		$this->path          = rtrim( $path . '/' );
+	}
+
+	/**
+	 * Registers Autoloader's autoload function.
+	 *
+	 * @throws \Exception When autoload_function cannot be registered.
+	 *
+	 * @param string $prefix Namespace prefix all classes have in common.
+	 * @param string $path   Path to the files to be loaded.
+	 */
+	public static function register_path( $prefix, $path ) {
+		$loader = new Autoloader( $prefix, $path );
```

Comment:
Maybe use `self` if the file gets renamed sometime?

```suggestion
		$loader = new self( $prefix, $path );
```